<launch>
    <arg name="camera_image_topic" default="/camera/image_raw"/>
    <arg name="camera_info_topic" default="/camera/camera_info"/>
    <arg name="warped_image_topic" default="/warped_image"/>
    <arg name="undistorted_image_topic" default="/undistorted_image"/>
    <arg name="homography_topic" default="/camera/camera_homography"/>
    <arg name="odometry_topic" default="/visual_odom/odom"/>
    <arg name="debug" default="false"/>
    <arg name="use_nodelet" default="false"/>
    <arg name="nodelet_manager" default=""/>

    <include file="$(find drive_ros_image_recognition)/launch/warp_image.launch">
        <arg name="camera_image_topic" default="$(arg camera_image_topic)"/>
        <arg name="camera_info_topic" default="$(arg camera_info_topic)"/>
        <arg name="warped_image_topic" default="$(arg warped_image_topic)"/>
        <arg name="undistort_out" default="$(arg undistorted_image_topic)"/>
        <arg name="homography_topic" default="$(arg homography_topic)"/>
        <arg name="debug" default="$(arg debug)"/>
        <arg name="use_nodelet" default="$(arg use_nodelet)"/>
        <arg name="nodelet_manager" default="$(arg nodelet_manager)"/>
    </include>

    <include file="$(find drive_ros_image_recognition)/launch/line_detection.launch">
        <arg name="incoming_image" default="$(arg warped_image_topic)"/>
        <arg name="camera_info" default="$(arg camera_info_topic)"/>
        <arg name="homography_topic" default="$(arg homography_topic)"/>
        <arg name="odometry_topic" default="$(arg odometry_topic)"/>
        <arg name="debug" default="$(arg debug)"/>
        <arg name="use_nodelet" default="$(arg use_nodelet)"/>
        <arg name="nodelet_manager" default="$(arg nodelet_manager)"/>
    </include>
</launch>
